packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -175,14 +180,11 @@ export const withBlockBindingSupport = createHigherOrderComponent(
 							if ( attributeName === 'url' ) {
 								attributes[ attributeName ] = null;
 							} else {
+								// Use the key of the source as a placeholder for the attribute.
+								// If it doesn't have a key, use the source label.
 								attributes[ attributeName ] =
-									source.getPlaceholder?.( {
-										registry,
-										context,
-										clientId,
-										attributeName,
-										args: bindings[ attributeName ].args,
-									} );
+									bindings[ attributeName ].args?.key ||
+									source.label;
```

Comment:
Here, I'm using just the `key` if it exists and the source label if it doesn't. The `key` has been reported to be too technical. From past conversations, there are other options to consider:

* Use always the source label. However, it could be confusing when there are multiple blocks connected to the same source but different keys. For example, I can see a common use case in templates.
* Use something more verbose like Post Meta "my_custom_field". The downside is that it could be too much.
* Add a new `label` property in the fields registration. This seems the best option to me, but it might take time so, if we decide to take that approach, I believe it could be done in a follow-up pull request.

I'd love to get more feedback on this.

Reply:
This has been moved [here](https://github.com/WordPress/gutenberg/pull/64910/files#diff-99d79bdba99fe4517413b0fd9503eef924e439c03d44504c1ad0606b727cebb6R164-R166) for sources without `getValues` and [here](https://github.com/WordPress/gutenberg/pull/64910/files#diff-cf4447b4134dbf24fa2819703999f0131449fb79a0611bd544c5e05fd80aa36bR23-R25) for post meta. But the discussion still applies.